src/general/NewGameSettings.cs

```diff
@@ -0,0 +1,638 @@
+                difficultyTab.Show();
+                difficultyTabButton.Pressed = true;
+                break;
+            case SelectedOptionsTab.Planet:
+                planetTab.Show();
+                planetTabButton.Pressed = true;
+                break;
+            case SelectedOptionsTab.Miscellaneous:
+                miscTab.Show();
+                miscTabButton.Pressed = true;
+                break;
+            default:
+                GD.PrintErr("Invalid tab");
+                break;
+        }
+
+        GUICommon.Instance.PlayButtonPressSound();
+        selectedOptionsTab = selection;
+    }
+
+    /*
+      GUI Control Callbacks
+    */
+
+    private void OnBackPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        Exit();
+    }
+
+    private bool Exit()
+    {
+        EmitSignal(nameof(OnNewGameSettingsClosed));
+        return true;
+    }
+
+    private void OnAdvancedPressed()
+    {
+        ProcessAdvancedSelection(true);
+    }
+
+    private void ProcessAdvancedSelection(bool playSound)
+    {
+        if (playSound)
+            GUICommon.Instance.PlayButtonPressSound();
+
+        basicOptions.Visible = false;
+        advancedButton.Visible = false;
+
+        advancedOptions.Visible = true;
+        basicButton.Visible = true;
+        tabButtons.Visible = true;
+    }
+
+    private void OnBasicPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        advancedOptions.Visible = false;
+        basicButton.Visible = false;
+        tabButtons.Visible = false;
+
+        advancedButton.Visible = true;
+        basicOptions.Visible = true;
+    }
+
+    private void OnConfirmPressed()
+    {
+        if (!isGameSeedValid)
+            return;
+
+        // Disable the button to prevent it being executed again.
+        confirmButton.Disabled = true;
+
+        settings.Difficulty = DifficultyPresetIndexToValue(difficultyPresetButton.Selected);
+        settings.Origin = LifeOriginIndexToValue(lifeOriginButton.Selected);
+        settings.Lawk = lawkButton.Pressed;
+        SetSeed(gameSeed.Text);
+
+        settings.MPMultiplier = mpMultiplier.Value;
+        settings.AIMutationMultiplier = aiMutationRate.Value;
+        settings.CompoundDensity = compoundDensity.Value;
+        settings.PlayerDeathPopulationPenalty = playerDeathPopulationPenalty.Value;
+        settings.GlucoseDecay = glucoseDecayRate.Value * 0.01;
+        settings.OsmoregulationMultiplier = osmoregulationMultiplier.Value;
+        settings.FreeGlucoseCloud = freeGlucoseCloudButton.Pressed;
+
+        settings.MapType = MapTypeIndexToValue(mapTypeButton.Selected);
+
+        settings.IncludeMulticellular = includeMulticellularButton.Pressed;
+        settings.EasterEggs = easterEggsButton.Pressed;
+
+        var scene = GD.Load<PackedScene>("res://src/general/MainMenu.tscn");
+        var mainMenu = (MainMenu)scene.Instance();
+        mainMenu.NewGameSetupDone(settings);
```

Comment:
This doesn't seem very good, there's now duplicate main menu instances and since this one isn't added to the scene tree, it will be orphaned, potentially causing a memory leak. Instead, I think it's better to just move the logic in `MainMenu.NewGameSetupDone` to here. It's pertaining to the new game settings anyways.